src/core/gltf2importer/gltf2parser.cpp

```diff
@@ -432,24 +432,47 @@ void GLTF2Parser::moveToThread(QThread *targetThread)
     QObject::moveToThread(targetThread);
     if (m_sceneRootEntity)
         m_sceneRootEntity->moveToThread(targetThread);
-    for (auto &anim : m_animators) {
-        anim.clip->moveToThread(targetThread);
-        anim.mapper->moveToThread(targetThread);
+
+    auto updateThread = [&](QObject* node) {
+        if(node && node->parent() == nullptr)
+            node->moveToThread(targetThread);
+    };
+
+    for (const TreeNode &treeNode : qAsConst(m_treeNodes)) {
+        updateThread(treeNode.entity);
+        for (Qt3DCore::QJoint * joint : treeNode.joints)
+            updateThread(joint);
     }
 
-    for (int i = 0, n = m_context->scenesCount(); i < n; i++) {
-        const Scene scene = m_context->scene(i);
-        const QVector<int> toRetrieveNodes = scene.rootNodeIndices;
-
-        for (int j = 0, m = toRetrieveNodes.size(); j < m; ++j) {
-            const TreeNode node = m_treeNodes[toRetrieveNodes.at(j)];
-            if (node.entity != nullptr)
-                node.entity->moveToThread(targetThread);
-            for (auto &joint : node.joints)
-                if (joint != nullptr)
-                    joint->moveToThread(targetThread);
-        }
+    for (AnimationDetails &anim : m_animators) {
+        updateThread(anim.clip);
+        updateThread(anim.mapper);
     }
+
+    for(Qt3DCore::QSkeleton* skeleton : m_skeletons) {
+        updateThread(skeleton);
+    }
+
+    iterateAssets<Mesh>([&] (const Mesh& mesh) {
```

Comment:
do you need to capture everything? (same below)

Reply:
done